concordium-cis2/src/lib.rs

```diff
@@ -813,6 +871,48 @@ pub enum Cis2Event<T: IsTokenId, A: IsTokenAmount> {
     TokenMetadata(TokenMetadataEvent<T>),
 }
 
+impl<T: IsTokenId, A: IsTokenAmount> schema::SchemaType for Cis2Event<T, A> {
+    fn get_type() -> schema::Type {
+        let mut event_map = BTreeMap::new();
+        event_map.insert(
+            EVENT_TAG_VECTOR[0],
+            (
+                "Transfer".to_string(),
+                schema::Fields::Unnamed(Vec::from([TransferEvent::<T, A>::get_type()])),
+            ),
+        );
+        event_map.insert(
+            EVENT_TAG_VECTOR[1],
+            (
+                "Mint".to_string(),
+                schema::Fields::Unnamed(Vec::from([MintEvent::<T, A>::get_type()])),
+            ),
+        );
+        event_map.insert(
+            EVENT_TAG_VECTOR[2],
+            (
+                "Burn".to_string(),
+                schema::Fields::Unnamed(Vec::from([BurnEvent::<T, A>::get_type()])),
+            ),
+        );
+        event_map.insert(
+            EVENT_TAG_VECTOR[3],
+            (
+                "UpdateOperator".to_string(),
+                schema::Fields::Unnamed(Vec::from([UpdateOperatorEvent::get_type()])),
+            ),
+        );
+        event_map.insert(
+            EVENT_TAG_VECTOR[4],
+            (
+                "TokenMetadata".to_string(),
+                schema::Fields::Unnamed(Vec::from([TokenMetadataEvent::<T>::get_type()])),
```

Comment:
I'm guessing this implementation will change based on the outcome of the discussion on slack?